src/pages/Swap/index.tsx

```diff
@@ -196,14 +197,27 @@ export default function Swap() {
   const [approval, approveCallback] = useApproveCallbackFromTrade(trade /* allowedSlippage */)
 
   // check if user has gone through approval process, used to show two step buttons, reset on token change
-  const [approvalSubmitted, setApprovalSubmitted] = useState<boolean>(false)
+  const [approvalsSubmitted, setApprovalsSubmitted] = useState<boolean[]>([])
+
+  const currentTradeIndex = useMemo(() => {
+    if (allPlatformTrades) {
+      return allPlatformTrades.findIndex(
+        trade => trade && platformOverride && trade.platform.name === platformOverride.name
+      )
```

Comment:
`currentTradeIndex` can be -1
- [`findIndex` returns -1](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) when no value is foudn.
- `trade` is always defined. 
- `platformOverride` is undefined by default.

Reply:
not sure what you are suggesting here?

Reply:
It should return 0 in both cases